src/index/IndexImpl.Text.cpp

```diff
@@ -116,26 +99,31 @@ void IndexImpl::logEntityNotFound(const string& word,
 }
 
 // _____________________________________________________________________________
-void IndexImpl::buildTextIndexFile(
-    const std::optional<std::pair<string, string>>& wordsAndDocsFile,
-    bool addWordsFromLiterals, TextScoringMetric textScoringMetric,
-    std::pair<float, float> bAndKForBM25) {
-  AD_CORRECTNESS_CHECK(wordsAndDocsFile.has_value() || addWordsFromLiterals);
+void IndexImpl::buildTextIndexFile(TextIndexConfig config) {
+  config.checkValid();
+  bool addWordsFromFiles =
+      config.docsFile_.has_value() &&
+      (config.wordsFile_.has_value() || config.useDocsFileForVocabulary_);
+  if (config.addOnlyEntitiesFromWordsFile_ &&
+      !config.useDocsFileForVocabulary_) {
+    throw std::runtime_error(
+        "The option to add entities from wordsfile can only be used together "
+        "with the option to build the text index from docsfile since "
+        "otherwise the wordsfile would be used anyway.");
+  }
```

Comment:
As this code only uses the `config`, it should be in member functions of the config to make the code here easier (especially the part that throws can be part of `checkValid` or something.